packages/babel-parser/src/parser/expression.js

```diff
@@ -1153,16 +1153,12 @@ export default class ExpressionParser extends LValParser {
         // https://tc39.es/proposal-private-fields-in-in
         // RelationalExpression [In, Yield, Await]
         //   [+In] PrivateIdentifier in ShiftExpression[?Yield, ?Await]
-        const start = this.state.start;
         const value = this.state.value;
         node = this.parsePrivateName();
         if (this.match(tt._in)) {
-          this.expectPlugin("privateIn");
           this.classScope.usePrivateName(value, node.start);
-        } else if (this.hasPlugin("privateIn")) {
-          this.raise(this.state.start, Errors.PrivateInExpectedIn, value);
         } else {
-          throw this.unexpected(start);
+          this.raise(this.state.start, Errors.PrivateInExpectedIn, value);
```

Comment:
Here we no longer throw at the private identifier, is it intentional?

Reply:
It's GitHub's diff not rendering it correctly; this was the deleted else-if branch.

Reply:
I don't think so. Before this PR `this.state.start` is accessed _before_ `parsePrivateName`. In this PR it is _after_. So the error positions are changed from

```
"throws": "Unexpected token (4:11)"
```

to

```
"SyntaxError: Private names are only allowed in property accesses (`obj.#x`) or in `in` expressions (`#x in obj`). (4:13)"
```

Reply:
The old code was
https://github.com/babel/babel/blob/bfcf783834cbe293ca540e84c1ad285ed2e728e2/packages/babel-parser/src/parser/expression.js#L1163
I didn't write this line, I just deleted the `hasPlugin` check and removed the now unreachable `else` clause.

But yeah, it's true that the old position without the plugin was better.